src/geode/geosciences_io/mesh/grdecl_input.cpp

```diff
@@ -0,0 +1,290 @@
+                if( !tokens.empty() )
+                {
+                    OPENGEODE_ASSERT( tokens.size() == 6,
+                        "[GRDECLInput::read_pillars] Wrong "
+                        "number of coordinates" );
+                    geode::Point3D top_point{ { geode::string_to_double(
+                                                    tokens[0] ),
+                        geode::string_to_double( tokens[1] ),
+                        geode::string_to_double( tokens[2] ) } };
+                    geode::Point3D bottom_point{ { geode::string_to_double(
+                                                       tokens[3] ),
+                        geode::string_to_double( tokens[4] ),
+                        geode::string_to_double( tokens[5] ) } };
+                    pillars[pillar_number++] = { std::move( top_point ),
+                        std::move( bottom_point ) };
+                }
+                std::getline( file_, line );
+            }
+            return pillars;
+        }
+
+        absl::FixedArray< double > read_depths()
+        {
+            absl::FixedArray< double > depths( 8 * nx_ * ny_ * nz_ );
+            auto line = geode::detail::goto_keyword( file_, "ZCORN" );
+            geode::index_t depths_number{ 0 };
+            std::getline( file_, line );
+            while( line != "/" )
+            {
+                const auto tokens = geode::string_split( line );
+                for( const auto depths_id : geode::Indices{ tokens } )
+                {
+                    depths[depths_number++] =
+                        geode::string_to_double( tokens[depths_id] );
+                }
+                std::getline( file_, line );
+            }
+            return depths;
+        }
+
+        std::array< geode::Point3D, 8 > cell_points(
+            const std::array< geode::index_t, 3 >& grid_coordinates,
+            absl::Span< const Pillar > pillars,
+            absl::Span< const double > depths ) const
+        {
+            const auto& pillars_id = cell_pillars_id( grid_coordinates );
+            const auto& bottom_left_pillar = pillars[pillars_id[0]];
+            const auto& top_left_pillar = pillars[pillars_id[1]];
+            const auto& bottom_right_pillar = pillars[pillars_id[2]];
+            const auto& top_right_pillar = pillars[pillars_id[3]];
+            std::array< geode::Point3D, 8 > points;
+            points[0] = interpolate_on_pillar(
+                depths[2 * grid_coordinates[0] + 4 * nx_ * grid_coordinates[1]
+                       + 4 * nx_ * ny_ * ( grid_coordinates[2] + 1 ) + 2 * nx_],
+                bottom_left_pillar );
+            points[1] = ( interpolate_on_pillar(
+                depths[2 * grid_coordinates[0] + 4 * nx_ * grid_coordinates[1]
+                       + 4 * nx_ * ny_ * ( grid_coordinates[2] + 1 ) + 2 * nx_
+                       + 1],
+                bottom_right_pillar ) );
```

Comment:
```suggestion
            points[1] = interpolate_on_pillar(
                depths[2 * grid_coordinates[0] + 4 * nx_ * grid_coordinates[1]
                       + 4 * nx_ * ny_ * ( grid_coordinates[2] + 1 ) + 2 * nx_
                       + 1],
                bottom_right_pillar );
```
to do for all points